components/up.libsonnet

```diff
@@ -176,31 +115,30 @@ local k = import 'ksonnet/ksonnet.beta.4/k.libsonnet';
         name: up.config.name,
         namespace: up.config.namespace,
       },
-    },
-  },
-
-  withServiceMonitor:: {
```

Comment:
I think we should keep `withServiceMonitor`. It is different than other changes, it adds an additional resource.
Also we have used the same pattern for every other component, so I personally prefer this pattern for external resources.

Reply:
I agree, we simply didn't want to match things. 
The funny thing is, we figured out a way to elegantly solve this too though. Maybe we can still improve it further down the road.

Essentially, if `serviceMonitor == true` then we create the object and add it to the overall object, else this serviceMonitor object is simply `null`.
Then if you look at the manifest generation, we check for object being `null` and then simply ignore them :)

WDYT?

Reply:
Fair enough 👍